tests/all/integration/api/session/test_session_api.py

```diff
@@ -0,0 +1,480 @@
+                },
+                "relationships": {
+                    "event": {"data": {"id": str(session.event_id), "type": "event"}},
+                    "track": {"data": {"id": str(track.id), "type": "track"}},
+                },
+            }
+        }
+    )
+
+    response = client.post(
+        '/v1/sessions', content_type='application/vnd.api+json', headers=jwt, data=data,
+    )
+
+    assert response.status_code == 201
+
+    session = Session.query.get(json.loads(response.data)['data']['id'])
+
+    assert session.title == 'Move Away'
+    assert session.subtitle == 'Moooove'
+    assert session.level == '456345678'
+    assert session.short_abstract == 'Speaking since birth'
+    assert session.complex_field_values is None
+
+
+def get_complex_custom_form_session(db, user):
+    session = get_minimal_session(db, user)
+    CustomForms(
+        event=session.event,
+        form='session',
+        field_identifier='slidesUrl',
+        type='text',
+        is_included=True,
+        is_required=True,
+    )
+    CustomForms(
+        event=session.event,
+        form='session',
+        field_identifier='bestFriend',
+        name='Best Friend',
+        type='text',
+        is_included=True,
+        is_required=True,
+        is_complex=True,
+    )
+    CustomForms(
+        event=session.event,
+        form='session',
+        field_identifier='transFatContent',
+        name='Trans Fat Content',
+        type='number',
+        is_included=True,
+        is_required=False,
+        is_complex=True,
+    )
+    db.session.commit()
+
+    return session
+
+
+def test_custom_form_complex_fields_missing_required(db, client, user, jwt):
+    session = get_complex_custom_form_session(db, user)
+
+    data = json.dumps(
+        {
+            'data': {
+                'type': 'session',
+                'id': str(session.id),
+                "attributes": {
+                    "title": "Move Away",
+                    "subtitle": "Moooove",
+                    "language": "English",
+                },
+            }
+        }
+    )
+
+    response = client.patch(
+        f'/v1/sessions/{session.id}',
+        content_type='application/vnd.api+json',
+        headers=jwt,
+        data=data,
+    )
+
+    db.session.refresh(session)
+
+    assert response.status_code == 422
+    assert json.loads(response.data) == {
+        'errors': [
+            {
+                'detail': "Missing required fields ['best_friend', 'slides_url']",
+                'source': {'pointer': '/data/attributes'},
+                'status': 422,
+                'title': 'Unprocessable Entity',
+            }
+        ],
+        'jsonapi': {'version': '1.0'},
+    }
+
+    assert session.title != 'Move Away'
+    assert session.subtitle != 'Moooove'
+    assert session.language is None
+
+
+def test_custom_form_create_complex_fields_missing_required(db, client, user, jwt):
```

Comment:
Codacy found an issue: [Function name "test_custom_form_create_complex_fields_missing_required" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5802755)